### PR TITLE
DOCO-704 Document Initial Drupal 10 Update Process

### PR DESCRIPTION
**Pull request by wordsmither**
Closes #DOCO-704

## Summary

**[Drupal 10 Migration Guides](https://pantheon.io/docs/drupal-10-migration)** - New guide sharing Pantheon's process for updating to Drupal 10.

## Effect

The following changes are already committed:

*
*

## Remaining Work and Prerequisites

The following changes still need to be completed:

- [ ] List any outstanding work here

### Dependencies and Timing

- [ ] Other prerequisites that must be completed before merging this PR

**Release**:
- [ ] When ready
- [ ] After date: $DATE

--------------------------------------------------

## Post Launch

**Do not remove** - To be completed by the docs team upon merge:

- [ ] Redirect `/docs/old-path/` => `/docs/new-path/` (if applicable)
- [ ] Include/exclude pages ^ respectively within docs search service provider (if applicable)
- [ ] For Heroes - add a props post to the [discussion board](https://discuss.pantheon.io/c/pantheon-platform/documentation/17).
- [ ] Remove from the [project board](https://github.com/pantheon-systems/documentation/projects/14)

### CODE REVIEW

**Comment by wordsmither** [comment]
[Preview](https://pr-7820--pantheon-docs.my.pantheonfrontend.website/docs/drupal-10-migration)

**Review by namespacebrian — CHANGES_REQUESTED** [review]
I sort of mis-characterized the steps I documented in a way that implied they could be used to migrate any site to Drupal 10, but these steps only cover freshly-created sites.  Migrating existing live D9 sites to D10 will involve more steps that we haven't documented yet, so I think we should re-frame this document as steps for creating a new Drupal 10 site on Pantheon.

I also omitted a step that should be explicitly called out: Step 1 is creating a new Drupal site using the "Drupal with Composer" upstream.

Requesting the following changes:
1. Change the document title and URL to position this as instructions for creating a new Drupal 10 site
2. Add two new steps to the beginning of the "Before You Begin" section:
   1. Create a new Drupal site using the "Drupal with Composer" CMS/Start State - attaching screenshots in case they are useful
   2. Clone the site locally with `git`

Dashboard: Create new site
Dashboard: Create new site, "Drupal with Composer" highlighted

File: source/content/drupal-10.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+<Alert title="Warning" type="danger" >
+
+These instructions were developed using beta versions of Drupal 10, and are for educational purposes only.  This page will be updated as production versions become available.
+
+</Alert>
+
+## Before You Begin
+
+This process requires that you are on PHP 8.1.  To upgrade:
```
Comment (namespacebrian) [inline source/content/drupal-10.md:20]: Might want to move this line to the 3rd step.

**Comment by whitneymeredith** [comment]
LGTM

**Pull request merged**